app/apollo/test/channel.fga.spec.js

```diff
@@ -0,0 +1,666 @@
+/**
+ * Copyright 2020 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { expect } = require('chai');
+const fs = require('fs');
+const { v4: UUID } = require('uuid');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+
+const { models } = require('../models');
+const resourceFunc = require('./api');
+const channelFunc = require('./channelApi');
+
+const apollo = require('../index');
+const { AUTH_MODEL } = require('../models/const');
+
+// If the current auth model does not support FGA, skip FGA unit tests without error.
+if (AUTH_MODEL === 'extauthtest' || AUTH_MODEL === 'passport.local') {
+  console.log(`Found non fine-grained auth model: ${AUTH_MODEL}. Skipping fine-grained auth tests.`);
+  return process.exit(0);
+}
+const externalAuth = require('../../externalAuth.js');
+const testHelperPath = externalAuth.ExternalAuthModels[AUTH_MODEL] ? externalAuth.ExternalAuthModels[AUTH_MODEL].testPath : `./testHelper.${AUTH_MODEL}`;
+const { prepareUser, prepareOrganization, signInUser } = require(testHelperPath);
+const testDataPath = externalAuth.ExternalAuthModels[AUTH_MODEL] ? externalAuth.ExternalAuthModels[AUTH_MODEL].testDataPath : `./app/apollo/test/data/${AUTH_MODEL}`;
+
+let mongoServer;
+let myApollo;
+
+const graphqlPort = 18000;
+const graphqlUrl = `http://localhost:${graphqlPort}/graphql`;
+const resourceApi = resourceFunc(graphqlUrl);
+const channelApi = channelFunc(graphqlUrl);
+
+let fgaToken01;
+let fgaToken02;
+
+let org01Data;
+let org01;
+
+let fgaUser01Data;
+let fgaUser02Data;
+
+const createOrganizations = async () => {
+  org01Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/cluster.spec.org_01.json`,
+      'utf8',
+    ),
+  );
+  org01 = await prepareOrganization(models, org01Data);
+};
+
+const createUsers = async () => {
+  fgaUser01Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user01.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser01Data);
+  fgaUser02Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user02.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser02Data);
+  return {};
+};
+
+const createChannels = async () => {
+  await models.Channel.create({
+    _id: 'fake_ch_id_1',
```

Comment:
```suggestion
    _id: UUID(),
```
Should this use a generated UUID (like in `createVersions`) so that tests and future devs don't attempt to use the `_id` attribute for matching etc?

Reply:
Same for other `_id`s below?

Reply:
Using a generated UUID for the purpose of preventing future devs from matching seems like a good change. I applied this in the createObjects section of each fga file (before unit tests)

Reply:
resolved